### PR TITLE
dstore/pmix: flock portability

### PR DESCRIPTION
**Pull request by karasevb**
Using the fcntl-locking instead of the flock

Cherry-picked from commit 3030a0c (PR #150)

### CODE REVIEW

**Comment by karasevb** [comment]
@artpol84 please rewiew

File: src/dstore/pmix_esh.c [hunk 1]
```diff
@@ -58,6 +58,14 @@ pmix_dstore_base_module_t pmix_dstore_esh_module = {
 #define ESH_ENV_NS_DATA_SEG_SIZE    "NS_DATA_SEG_SIZE"
 #define ESH_ENV_LINEAR              "SM_USE_LINEAR_SEARCH"
 
+#define _ESH_LOCK(lockfd, operation) \
+do{ \
+    struct flock fl = {0}; \
+    fl.l_type = operation; \
+    fl.l_whence = SEEK_SET; \
+    fcntl(lockfd, F_SETLKW, &fl); \
```
Comment (jjhursey) [inline src/dstore/pmix_esh.c]: Please check the return value of the `fcntl` function. Coverity will flag this, and it is good to check anyway.

**Comment by artpol84** [comment]
yeah, this was discovered by coverity yesterday. Boris is addressing this.

**Comment by artpol84** [comment]
We need to fix this in the master first

**Comment by artpol84** [comment]
@jjhursey as this PR was merged into the master as is I suggest to preserve the sequence of PR's in the v2.x.
So we will open a new PR to the master addressing this and then ASAP PR to v2.x

**Comment by jjhursey** [comment]
Ok. I'm fine with that then. I kind of want to see the PR to `master` first just so we don't lose it in the shuffle.

**Comment by artpol84** [comment]
Sure.
@karasevb please, PR ASAP.

**Pull request merged**